### DISCUSSION TITLE
Generic 2.4" TFT Touch Display + ESP32

### DISCUSSION THREAD

**Discussion by AlexLucardGoodman** [Moddable-OpenSource/moddable#1037] [Q&A]
Hello again!
Hope everyone is doing well. 
I want to test a 2.4inch touch display with an esp32. I came across this exact one which also comes with a SD card reader: http://www.lcdwiki.com/2.4inch_Arduino_Display. Is it possible to use this shield with the moddable sdk? 
The wiring guide https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/displays/wiring-guide-generic-2.4-spi-esp32.md shows different pin names than the ones in the shield.
Can this screen be used??? Ive seen some different esp32 wirings and im confused.
Thanks in advance.

**Comment by phoddie** [comment]
Howdy.

This screen has an 8-bit SPI interface. which is not currently supported. There's some indication it might also work with 1-bit SPI, but hard to say from the site.

**Reply by AlexLucardGoodman** [reply]
Thanks for your answer. Would it work with 1bit spi in the moddable environment? any ideas how to go about it?

**Reply by phoddie** [reply]
If the display can operate in 1-bit SPI mode, it should be straightforward to get it working. I cannot say with confidence whether or not that display supports 1-bit SPI. The ST7789 display controller it uses does support 1-bit SPI, so it is a question of whether the SPI mode can be configured and the required pins are accessible.

Assuming that is the case, you just need to create manifest with the correct configuration for that display. You could start by modifying the `esp32/moddable_zero` [manifest](https://github.com/Moddable-OpenSource/moddable/blob/3c3f72f2b12775481e2962222eeb2dfc2321890a/build/devices/esp32/targets/moddable_zero/manifest.json#L15-L25) for your wiring. Note that the ILI9341 driver it uses also works with the ST7789, You may also need to change the register configuration for your display. The AtomS3 manifest has a [good example](https://github.com/Moddable-OpenSource/moddable/blob/3c3f72f2b12775481e2962222eeb2dfc2321890a/build/devices/esp32/targets/atoms3/manifest.json#L52-L82) of that.

**Reply by AlexLucardGoodman** [reply]
Excellent explanation. Ill be trying this and will write back when i get some results. Thank you, Phoddie.

**Comment by AlexLucardGoodman** [comment]
So, i haven been able to make this screen work yet. Ive come across another one and wanted to make sure if this one might work in order to go on with some PIU tests. 
Its a 2.8 inches Touch St7735 240x320 Tft. The shop link to the item is: https://articulo.mercadolibre.com.ar/MLA-1167011096-pantalla-28-pulgadas-con-touch-arduino-st7735-240x320-tft-_JM#reco_item_pos=2&reco_backend=machinalis-seller-items-pdp&reco_backend_type=low_level&reco_client=vip-seller_items-above&reco_id=3a83aee0-3c90-4722-bb64-569221c6d5ee
Its supposed to be the red version shown in the first picture. 
Might this work with the moddable sdk? 
thank you!

**Reply by phoddie** [reply]
Maybe? It looks like 4-wire SPI. If so, there's a good chance. I can't find an image that's clear enough to read the pin labels to say for sure.

If you don't mind me saying, I really like [this device](https://www.moddable.com/moddable-two). I'm 100% sure it works with the Moddable SDK. ;)

**Reply by AlexLucardGoodman** [reply]
Hi Peter, sorry for the delay in my reaply. Ive been trying to contact the seller to get the pin names in order to see if it would work but havent been able to yet.
Moddable Two is a pure beauty and would love to get my hands on one of those sometime in the near future.
Still, what im trying to achieve here is test the Moddable SDK with the low cost electronic parts available in south america. Theres many problems here regarding imports and we dont have many options, sadly...
If its ok, ill write back in this same thread when I find a touch screen that might work to hear your oppinion.
Thank you, as always.

**Reply by phoddie** [reply]
@AlexLucardGoodman – understood. Regional differences in product availability are unfortunate. Alas, international shipping is complex and expensive. Good luck!
